### PR TITLE
chore: add `into_inner()` for ICS07 `ConsensusState`

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
## Description
- Adds an `into_inner()` method for the ICS07 `ConsensusState`, so that dependent projects won't need to perform redundant cloning.
- Removes redundant cloning within the IC07 `check_substitute` implementation.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: docs/architecture/adr-007-light-client-contexts.md [hunk 1]
```diff
@@ -214,9 +214,7 @@ trait ClientState<ClientValidationContext, ClientExecutionContext>
 
 The problem with defining all methods directly under `ClientState` is that it would force users to use fully qualified notation to call any method.
 
-This arises from the fact that no method uses both generic parameters. [This playground] provides an explanatory example. Hence, our solution is to have all methods in a trait use every generic parameter of the trait to avoid this problem.
-
-[This playground]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=da65c22f1532cecc9f92a2b7cb2d1360
```
Comment (seanchen1991) [inline docs/architecture/adr-007-light-client-contexts.md]: Why remove this playground link?

File: ibc-clients/ics07-tendermint/src/client_state/validation.rs [hunk 2]
```diff
@@ -249,11 +249,11 @@ where
         upgrade_path: substitute_upgrade_path,
     } = substitute_client_state;
 
-    (subject_trust_level == substitute_trust_level
-        && subject_unbonding_period == substitute_unbonding_period
-        && subject_max_clock_drift == substitute_max_clock_drift
-        && subject_proof_specs == substitute_proof_specs
-        && subject_upgrade_path == substitute_upgrade_path)
+    (subject_trust_level == &substitute_trust_level
```
Comment (seanchen1991) [inline ibc-clients/ics07-tendermint/src/client_state/validation.rs:252]: Good catch on this and the extraneous `clone`!

**Review by seanchen1991 — APPROVED** [review]
Looks good in general, just a question about the removed playground link.

File: docs/architecture/adr-007-light-client-contexts.md (hunk 1 above)
Comment (Farhad-Shabani) [reply docs/architecture/adr-007-light-client-contexts.md]: The link is not working anymore and breaks the `md-link-check` job. Not sure if there is a way to recover that.

File: ibc-clients/ics07-tendermint/src/client_state/validation.rs (hunk 2 above)
Comment (Farhad-Shabani) [reply ibc-clients/ics07-tendermint/src/client_state/validation.rs:252]: This caught my eye while integrating the client recovery into the Sovereign light client.

File: docs/architecture/adr-007-light-client-contexts.md (hunk 1 above)
Comment (seanchen1991) [reply docs/architecture/adr-007-light-client-contexts.md]: Weird, the link works for me. But if it's breaking the link check, I don't think it's worth figuring out why; easier to just remove it 👍

Comment (Farhad-Shabani) [reply docs/architecture/adr-007-light-client-contexts.md]: Gave it another try. Yea, sounds like it's working now. Reverted that! Not sure what was the issue.

**Pull request merged**